### PR TITLE
56 fix eegrasp docstrings

### PR DESCRIPTION
**Pull request by julioRodino**
Fix docstrings of the EEGrasp class.

### CODE REVIEW

File: eegrasp/eegrasp.py [hunk 1]
```diff
@@ -27,10 +31,15 @@ def __init__(self, data=None, coordinates=None, labels=None):
         """
         Parameters
         ----------
-        data: 2-d array, where the first dim are channels and the second is
-        samples.  
-        Coordenates: ndim array with position of the electrodes.
-        labels: 1-d array with the channel names.
+        data : ndarray | mne object | None.
```
Comment (aweinstein) [inline eegrasp/eegrasp.py:34]: Specify which MNE object.

File: eegrasp/eegrasp.py [hunk 2]
```diff
@@ -27,10 +31,15 @@ def __init__(self, data=None, coordinates=None, labels=None):
         """
         Parameters
         ----------
-        data: 2-d array, where the first dim are channels and the second is
-        samples.  
-        Coordenates: ndim array with position of the electrodes.
```
Comment (aweinstein) [inline eegrasp/eegrasp.py:32]: Change coordenates by coordinates

File: eegrasp/eegrasp.py [hunk 3]
```diff
@@ -27,10 +31,15 @@ def __init__(self, data=None, coordinates=None, labels=None):
...
-        labels: 1-d array with the channel names.
+        data : ndarray | mne object | None.
+            2D array. Where the first dim are channels and the second is samples. If 3D, the first
+            dimension is trials. If an mne object is passed, the data will be extracted from it
+            along with the coordinates and labels of the channels. If None, the class will be
```
Comment (aweinstein) [inline eegrasp/eegrasp.py:37]: Change None to `None`.

File: eegrasp/eegrasp.py [hunk 4]
```diff
@@ -27,10 +31,15 @@ def __init__(self, data=None, coordinates=None, labels=None):
...
+            initialized without data. 
+        Coordenates : ndarray | list | None.
+            N-dim array with position of the electrodes.
```
Comment (aweinstein) [inline eegrasp/eegrasp.py:40]: Explain how the different cases are handled.

File: eegrasp/eegrasp.py [hunk 5]
```diff
@@ -27,10 +31,15 @@ def __init__(self, data=None, coordinates=None, labels=None):
...
+        labels : ndarray | list | None.
```
Comment (aweinstein) [inline eegrasp/eegrasp.py:41]: Explain how the different cases are handled.

File: eegrasp/eegrasp.py [hunk 6]
```diff
@@ -46,13 +55,14 @@ def euc_dist(self, pos):
...
+        pos : ndarray.
+            2d or 3d array of channels by dimensions
 
         Returns
         -------
-        output: 2d array of channels by channels with the euclidean distance.
-        description: compute the euclidean distance between every channel in
-        the array
+        output: ndarray.
+            Dimension of the array is Channels by Channels with the euclidean distance between
```
Comment (aweinstein) [inline eegrasp/eegrasp.py:64]: by Channels you mean number of channels?

File: eegrasp/eegrasp.py [hunk 7]
```diff
@@ -71,27 +81,35 @@ def gaussian_kernel(self, x, sigma=0.1):
 
         Notes
         -----
-        This function is supposed to be used in the PyGSP2 module but
-        is repeated here since there is an error in the available version
-        of the toolbox (03/04/2024 dd/mm/yyyy)
+        This function is supposed to be used in the PyGSP2 module but is repeated here since there
```
Comment (aweinstein) [inline eegrasp/eegrasp.py:84]: This bug doesn't exist in PyGSP anymore. We should remove the function (but in a different issue/pr).

File: eegrasp/eegrasp.py [hunk 8]
```diff
@@ -71,27 +81,35 @@ def gaussian_kernel(self, x, sigma=0.1):
...
         return np.exp(-np.power(x, 2.) / (2.*np.power(float(sigma), 2)))
 
     def compute_distance(self, coordinates=None, method='Euclidean', normalize=True):
         """
-        Method for computing the distance.
+        Computing the distance based on electrode coordinates.
+
+        Parameters
+        ----------
+        coordinates : ndarray | None.
```
Comment (aweinstein) [inline eegrasp/eegrasp.py:102]: What happens when it is None?

File: eegrasp/eegrasp.py [hunk 9]
```diff
@@ -71,27 +81,35 @@ def gaussian_kernel(self, x, sigma=0.1):
...
+            N-dim array with position of the electrodes.
+        method : string.
+            Options are: 'Euclidean'. Method used to compute the distance matrix.
+        normalize : bool.
```
Comment (aweinstein) [inline eegrasp/eegrasp.py:106]: What happens if it is false?

File: eegrasp/eegrasp.py [hunk 10]
```diff
@@ -115,18 +133,20 @@ def compute_graph(self, W=None, epsilon=.5, sigma=.1):
...
-        W -> if W is passed, then the graph is computed. 
-        Otherwise the graph will be computed with self.W.
-        W should correspond to a non-sparse 2-D array.
-        Epsilon -> maximum distance to threshold the array.
-        sigma -> Sigma parameter for the gaussian kernel.
-
-        method: NN -> Nearest Neighbor
-                Gaussian -> Gaussian Kernel used based on the self.W matrix
+        W : numpy ndarray. 
+            if W is passed, then the graph is computed. Otherwise the graph will be computed with
```
Comment (aweinstein) [inline eegrasp/eegrasp.py:137]: Add the other possible value to`W`.

File: eegrasp/eegrasp.py [hunk 11]
```diff
@@ -115,18 +133,20 @@ def compute_graph(self, W=None, epsilon=.5, sigma=.1):
...
+            `self.W`. `W` should correspond to a non-sparse 2-D array.
+        epsilon : float.
+            Maximum distance to threshold the array.
```
Comment (aweinstein) [inline eegrasp/eegrasp.py:140]: Better: "Any distance smaller than `epsilon` is set to zero".

File: eegrasp/eegrasp.py [hunk 12]
```diff
@@ -115,18 +133,20 @@ def compute_graph(self, W=None, epsilon=.5, sigma=.1):
...
+        sigma : float.
+            Sigma parameter for the gaussian kernel.
+        method: string. 
+            Options are: "NN" or "Gaussian". Nearest Neighbor or Gaussian Kernel used based on the
+            `self.W` matrix respectively depending on the method used.
 
         Returns
         -------
-        G: Graph structure from PyGSP2        
+        G: Graph structure from PyGSP2.
```
Comment (aweinstein) [inline eegrasp/eegrasp.py:149]: Better: "PyGSP2 graph structure"

Is not the same as a NetworkX structure?

File: eegrasp/eegrasp.py [hunk 13]
```diff
@@ -155,6 +175,20 @@ def compute_graph(self, W=None, epsilon=.5, sigma=.1):
     def interpolate_channel(self, graph=None, data=None, missing_idx=None):
         """
         Interpolate missing channel.
+
+        Parameters
+        ----------
+        graph : PyGSP2 Graph | None.
```
Comment (aweinstein) [inline eegrasp/eegrasp.py:181]: What happen whe is None?

File: eegrasp/eegrasp.py [hunk 14]
```diff
@@ -155,6 +175,20 @@ def compute_graph(self, W=None, epsilon=.5, sigma=.1):
...
+            Graph structure from PyGSP2.
+        data : ndarray | None.
+            2d array of channels by samples.
```
Comment (aweinstein) [inline eegrasp/eegrasp.py:184]: What happen whe is None?

File: eegrasp/eegrasp.py [hunk 15]
```diff
@@ -155,6 +175,20 @@ def compute_graph(self, W=None, epsilon=.5, sigma=.1):
...
+        missing_idx : int | None.
```
Comment (aweinstein) [inline eegrasp/eegrasp.py:185]: What happen whe is None?

File: eegrasp/eegrasp.py [hunk 16]
```diff
@@ -222,16 +260,25 @@ def fit_epsilon(self, data=None, distances=None, sigma=0.1,
 
         Parameters
         ----------
-        distances -> Unthresholded distance matrix (2-dimensional array).
-        It can be passed to the instance of
-        the class or as an argument of the method.
-        sigma -> parameter of the Gaussian Kernel transformation
+        data : ndarray | None.
```
Comment (aweinstein) [inline eegrasp/eegrasp.py:263]: What happen whe is None?

File: eegrasp/eegrasp.py [hunk 17]
```diff
@@ -222,16 +260,25 @@ def fit_epsilon(self, data=None, distances=None, sigma=0.1,
...
+        distances : ndarray | None.
```
Comment (aweinstein) [inline eegrasp/eegrasp.py:264]: What happen whe is None?

File: eegrasp/eegrasp.py [hunk 18]
```diff
@@ -310,6 +357,21 @@ def fit_sigma(self, data=None, distances=None, epsilon=0.5,
 
         Parameters
         ----------
+        data : ndarray | None.
```
Comment (aweinstein) [inline eegrasp/eegrasp.py:360]: What happen whe is None?

File: eegrasp/eegrasp.py [hunk 19]
```diff
@@ -310,6 +357,21 @@ def fit_sigma(self, data=None, distances=None, epsilon=0.5,
...
+            2d array of channels by samples.
+        distances : ndarray | None.
```
Comment (aweinstein) [inline eegrasp/eegrasp.py:362]: What happen whe is None?

**Review by aweinstein — CHANGES_REQUESTED** [review]

File: eegrasp/eegrasp.py (hunk 1 above)
Comment (julioRodino) [reply eegrasp/eegrasp.py:34]: Specified 3 mne objects. This covers the uses for averaged epochs or erps, epochs and continuous data which covers the analysis of resting state data.

Comment (aweinstein) [reply eegrasp/eegrasp.py:34]: I don't see the change.

File: eegrasp/eegrasp.py (hunk 6 above)
Comment (julioRodino) [reply eegrasp/eegrasp.py:64]: yes

Comment (aweinstein) [reply eegrasp/eegrasp.py:64]: Then say so. Even better, use something linked to variables relevant to the code, like `len(something)` by `len(something)`

File: eegrasp/eegrasp.py (hunk 1 above)
Comment (julioRodino) [reply eegrasp/eegrasp.py:34]: Haven't made the commit yet

File: eegrasp/eegrasp.py (hunk 8 above)
Comment (julioRodino) [reply eegrasp/eegrasp.py:108]: If `coordinates = None` the self.coordinates will be used to compute the distance. If, in that case, `self.coordinates = None` then there is nothing to compute :)

File: eegrasp/eegrasp.py (hunk 12 above)
Comment (julioRodino) [reply eegrasp/eegrasp.py:158]: Nop, they are different. Though it is possible to convert a Pygsp2 graph object to a NetworkX structure

File: eegrasp/eegrasp.py (hunk 10 above)
Comment (julioRodino) [reply eegrasp/eegrasp.py:146]: Done

**Review by aweinstein — APPROVED** [review]

**Pull request merged**